presto-testng-services/src/main/java/io/prestosql/testng/services/StackTraceUtil.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.testng.services;
+
+import java.lang.management.LockInfo;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.time.Duration;
+
+public final class StackTraceUtil
+{
+    private StackTraceUtil() {}
+
+    /**
+     * The header section of the thread
+     * @param threadInfo the thread information
+     * @return the string builder of the header starting at a new line.
+     */
```

Comment:
By the rule of thumb we avoid comments. We try to comment that are part of SPI or things that do complex and unexpected things.